### ISSUE TITLE
ShouldAllBeEquivalentTo, throws FormatException on not null properties in 4.19.3 but works in 4.19.2

### ISSUE THREAD

**Issue by DerAlbertCom** [fluentassertions/fluentassertions#619]
The following xUnit Test fails (on intend), this is the reproduction version of a FA bug.

```C#
        [Fact]
        public void Failing_Test()
        {
            var inputClaims = new[]
            {
                new Claim("A1", "FooA1"),
                new Claim("A2", "FooA2"),
                new Claim("foo:C1", "FooC1"),
                new Claim("foo:C2", "FooC2"),
                new Claim("bar:C1", "BarC1"),
                new Claim("bar:C2", "BarC2"),
            };
            var claimsPrincipal = new ClaimsPrincipal(new ClaimsIdentity(inputClaims));
            var filter = claimsPrincipal.Claims.Where(c => c.Type.StartsWith("A")).ToList();

            filter.ShouldAllBeEquivalentTo(new[]
            {
                new Claim("A1", "FooA1"),
                new Claim("A2", "FooA2")
            });
        }
```
throws the exception in Version 4.19.3
```
System.FormatException: Input string was not in a correct format.
   at System.Text.StringBuilder.FormatError()
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at FluentAssertions.Execution.MessageBuilder.FormatArgumentPlaceholders(String failureMessage, Object[] failureArgs) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\MessageBuilder.cs:line 66
   at FluentAssertions.Execution.MessageBuilder.Build(String message, Object[] messageArgs, String reason, ContextDataItems contextData) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\MessageBuilder.cs:line 42
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args) in C:\projects\fluentassertions-vf06b\Src\Core\Execution\AssertionScope.cs:line 190
   at FluentAssertions.Equivalency.EnumerableEquivalencyValidator.LooselyMatchAgainst[T](IList`1 subjects, Object expectation, Int32 expectationIndex) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EnumerableEquivalencyValidator.cs:line 125
   at FluentAssertions.Equivalency.EnumerableEquivalencyValidator.AssertElementGraphEquivalency[T](T[] subjects, Object[] expectations) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EnumerableEquivalencyValidator.cs:line 79
   at FluentAssertions.Equivalency.EnumerableEquivalencyValidator.Execute[T](T[] subject, Object[] expectation) in C:\projects\fluentassertions-vf06b\Src\Core\Equivalency\EnumerableEquivalencyValidator.cs:line 39
```
give a correct message in Version 4.19.2 (shortened)
```
Xunit.Sdk.XunitException: Expected item[0].Subject to be <null>, but found 

System.Security.Claims.ClaimsIdentity
{
   Actor = <null>
   AuthenticationType = <null>
   BootstrapContext = <null>
   Claims = {A1: FooA1, A2: FooA2, foo:C1: FooC1, foo:C2: FooC2, bar:C1: BarC1, bar:C2: BarC2}
   ExternalClaims = 
   IsAuthenticated = False
   Label = <null>
   Name = <null>
   NameClaimType = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name"
   RoleClaimType = "http://schemas.microsoft.com/ws/2008/06/identity/claims/role"
}.
```
Would love to have 4.19.4  with the message of 4.19.2 ;) Seem similar to #613

**Comment by SaroTasciyan** [comment]
You mean `v4.19.x` , am I correct?

**Comment by DerAlbertCom** [comment]
of course.

**Comment by dennisdoomen** [comment]
Fixed in 4.19.4
